lang/csharp/src/apache/main/Specific/ObjectCreator.cs

```diff
@@ -110,16 +110,26 @@ public override int GetHashCode()
         /// <param name="name">the object type to locate</param>
         /// <param name="throwError">whether or not to throw an error if the type wasn't found</param>
         /// <returns>the object type, or <c>null</c> if not found</returns>
-        private Type FindType(string name,bool throwError)
+        private Type FindType(string name, bool throwError)
         {
-            Type type;
+            var isList = false;
+            var isNullable = false;
+
+            const string listPrefix = "IList<";
+            if (name.StartsWith(listPrefix) && name.EndsWith(">"))
+            {
+                name = name.Substring(listPrefix.Length, name.Length - listPrefix.Length - 1);
+                isList = true;
+            }
+
+            const string nullablePrefix = "Nullable<";
+            if (name.StartsWith(nullablePrefix) && name.EndsWith(">"))
```

Comment:
So this means you can have `IList<Nullable<SomeType>>`, but the code below returns either a List or a Nullable, so maybe it should be an `else if` here?

Reply:
yeah, i don't know if that can ever be hit, changed.